src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -194,11 +194,14 @@ private bool TryReplaceBannedFunction(FunctionExpression expression, [NotNullWhe
                     throw new ArgumentException($"Expected 0 properties for binary function {expression.Function}");
                 }
 
+                // Need to copy the banned operator or we may get a duplicate key exception when creating the syntax hierarchy.
```

Comment:
>duplicate key exception [](start = 68, length = 23)

The duplicate key exception could mean that we have reused a SyntaxHierarchy object that was populated with data from one AST with another AST that reuses parts of the old AST. That class was never designed for that. A whole new SyntaxHierarchy must be created for every new AST.

Reply:
That's not the case here because we are creating a new `SyntaxHierarchy` for each new AST. The duplicate key exception occurs if a JSON template contains more than one the same functions that can be replaced by an operator, for example, `"[add(1, 1)]"`. The the AST will contain multiple `+` operator syntaxes referencing to the same object if we don't do the copy. Calling `parentMap.Add(node, parent)` in `SyntaxHierarchy` will cause the duplicate key exception.

Reply:
Good find. Would you mind fixing that? I think the fix would just be to turn `BannedBinaryOperatorLookup` into a method with a `switch` statement so that it generates a new token on each call rather than storing and reusing them.

Reply:
Yeah I think that's better than cloning the cached value since it might be used somewhere else. I'll fix it.

Reply:
Changed the dictionary to a method.